### PR TITLE
refactor(space): delete completion-action pipeline + consolidate approval banners (PR 4/5)

### PR DESCRIPTION
**Pull request by lsm**
Implements PR 4/5 of `docs/plans/remove-completion-actions-task-agent-as-post-approval-executor.md`.

## Summary

- Deletes the completion-action runtime pipeline: executors, pending-action pre-commit, RPC intercept, `approve_completion_action` MCP tool, and `PendingCompletionActionBanner`. After PR 2/5, every completion path routes through `PostApprovalRouter`; these are now dead code.
- Consolidates `SpaceTaskPane` banners into a single-slot renderer driven by `resolveActiveTaskBanner(task, gates?)` in `lib/task-banner.ts`. Precedence: `blocked > post_approval_blocked > task_completion_pending > gate_pending > null`.
- Simplifies `workflow-autonomy.ts` to `isWorkflowAutoClosingAtLevel(wf, level)` — a single check against `workflow.completionAutonomyLevel` replaces the old per-node `requiredLevel` survey.
- Trims the remaining four banners (`TaskBlockedBanner`, `PendingGateBanner`, `PendingTaskCompletionBanner`, `PendingPostApprovalBanner`) to the one-line rule by composing the shared `InlineStatusBanner` primitive.
- DB schema and shared types are deliberately untouched — those are deferred to PR 5/5.

## Test plan

- [x] `./scripts/test-daemon.sh` — 11113 passing, 0 failing
- [x] `bunx vitest run` (web) — 7231 passing, 0 failing
- [x] `cd packages/shared && bun test tests/workflow-autonomy.test.ts` — 4 passing
- [x] `bun run check` — lint + typecheck + knip + session-guards clean
- [x] `post-approval-banner-minimal.e2e.ts` — fully implemented (single-line height + one-banner invariant)
- [x] `post-approval-no-route.e2e.ts` — fully implemented (standalone approved task shows no post-approval banner)
- [ ] `post-approval-merge-autonomy-high.e2e.ts` — skipped stub (needs mock `gh` + LLM mock — see file header)
- [ ] `post-approval-merge-autonomy-low.e2e.ts` — skipped stub (same infra)
- [ ] `post-approval-merge-human-rejects.e2e.ts` — skipped stub (same infra)

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
PR 4/5 is well-executed. All required deletions are complete and the refactored surfaces are correct. Two minor issues to address before merge.

---

## Issues

### 1. Stale comments in `SpaceTaskPane.tsx` and `TaskStatusActions.tsx` (cosmetic but misleading)

`SpaceTaskPane.tsx` lines 37–38:
```
// task-agent-as-post-approval-executor refactor. Not yet written by any
// runtime path — PR 2 wires it in.
```
`TaskStatusActions.tsx` line 13:
```
// produces this status yet — PR 2 wires it in.
```
Line 42:
```
// PR 1/5 schema-only: `approved` has no user-facing enter edge yet.
```

These comments were left over from PR 1/5 and are factually wrong as of PR 4/5. The `approved` status is now actively written by `dispatchPostApproval` and the post-approval router. A future reader following these comments will chase a ghost. Please update them to reflect current state.

### 2. `post-approval-no-route.e2e.ts` — second test's assertion of `approved` status may be flaky in a live daemon environment

The second test in `post-approval-no-route.e2e.ts` (lines 139–169) manually drives a standalone task to `approved` via a raw `spaceTask.update` RPC, then immediately asserts `task.status === 'approved'` via a follow-up RPC read. This bypasses `dispatchPostApproval` (which is only triggered by `approve_task` tool calls, not by the generic update path), so the task stays at `approved` deterministically in isolation.

However, the test comment (lines 142–145) says _"there is no router run — the task stays in `approved` until someone explicitly transitions it"_ but the preceding test (`standalone task at approved`) navigates to the same task URL and asserts all banners are hidden. If `dispatchPostApproval` were ever called on that path — e.g. if the daemon's reconciliation loop runs a tick on a freshly-set `approved` task — the status could advance to `done` between the RPC write and the UI assertion, causing a false-positive pass (banner never appears anyway for a `done` task) or a flaky assertion on the second test.

This is currently not a bug because the update path does not call `dispatchPostApproval`, but the test has no comment or guard making that dependency explicit. A one-line comment noting _"direct spaceTask.update does not trigger the router — task stays at approved"_ would prevent this from being misunderstood or accidentally broken by a future refactor.

---

## What was reviewed and confirmed correct

**Shared:**
- `workflow-autonomy.ts`: old exports (`EMPTY_ACTIONS_AUTONOMY_THRESHOLD`, `isAutonomousWithoutActions`, `BlockingAction`, `BlockingWorkflow`, `AutonomousWorkflowCount`, `isWorkflowAutonomousAtLevel`, `countAutonomousWorkflows`, `collectCompletionActions`) are fully deleted; file contains only `isWorkflowAutoClosingAtLevel`. Logic `level >= (wf.completionAutonomyLevel ?? 5)` is correct.
- `workflow-autonomy.test.ts`: 4 tests covering `completionAutonomyLevel` at levels 1–5, explicit 5, explicit 1, and missing (defaults to 5). No references to deleted exports. All 4 pass.

**Web:**
- `task-banner.ts`: `ActiveBanner` discriminated union is correct. Precedence `blocked > post_approval_blocked > task_completion_pending > gate_pending > null` is implemented exactly as specified. The `post_approval_blocked` branch correctly requires `task.status === 'approved'` AND a non-empty (trimmed) `task.postApprovalBlockedReason`. `completion_action` checkpoints fall through to `null` as documented.
- `task-banner.test.ts`: 16 tests covering all precedence combos, the `post_approval_blocked` whitespace edge cases, the `completion_action` legacy passthrough, all gate loading states, and standalone task (no runId). All 16 pass.
- `SpaceTaskPane.tsx`: old 3-conditional banner stack correctly replaced by single `resolveActiveTaskBanner` call. All four banner components imported and rendered correctly under the switch. Gate summaries correctly passed. No `PendingCompletionActionBanner` import.
- `TaskStatusActions.tsx`: no `PendingCompletionActionBanner` import or render. The `pendingCheckpointType` prop no longer includes `'completion_action'` as a valid filter value.
- `PendingCompletionActionBanner.tsx`: confirmed deleted (absent from directory listing).
- `PendingGateBanner.tsx`: composes `InlineStatusBanner`, one-line shape for each gate row.
- `PendingTaskCompletionBanner.tsx`: composes `InlineStatusBanner` for the status row; modals are separate (not inline multi-line), satisfying the one-line rule.
- `TaskBlockedBanner.tsx`: composes `InlineStatusBanner`. `data-reason` propagated via `dataAttrs`. The `data-tone` attribute is emitted by `InlineStatusBanner` on line 119 — the E2E assertions on `data-tone='red'` and `data-reason='execution_failed'` will work correctly.
- `PendingPostApprovalBanner.tsx`: composes `InlineStatusBanner`, error rendered as a separate element outside the banner (not inline), one-line rule satisfied.
- `AutonomyWorkflowSummary.tsx`: correctly imports and calls `isWorkflowAutoClosingAtLevel` from `@neokai/shared`. No references to deleted exports. 5 unit tests pass.
- `SpaceOverview.tsx` and `SpaceSettings.tsx`: both use `AutonomyWorkflowSummary` component; no direct calls to deleted helpers.
- No `completionActions` field references remain in `packages/web/src/` beyond comments explaining the deletion.
- No `__tests__/` file references deleted components.

**E2E:**
- `post-approval-banner-minimal.e2e.ts`: fully implemented. Asserts single-line height (`≤ 44px`) via `boundingBox()` and the one-banner invariant via test-id counting. Two live test cases.
- `post-approval-no-route.e2e.ts`: fully implemented. Asserts no banner for `approved` standalone task; asserts `task.status === 'approved'` via RPC. Two live test cases.
- Three skipped stubs (`post-approval-merge-autonomy-high`, `post-approval-merge-autonomy-low`, `post-approval-merge-human-rejects`): all use `test.describe.skip` with a detailed rationale comment listing the missing infra (mock `gh`, LLM mock, per-spec env override). Stubs are not empty — each contains a commented step-by-step outline. Correctly skipped.

**Review by lsm — COMMENTED** [review]
## Daemon-side review: PR 4/5 deletion of completion-action pipeline

All tests pass (2446 daemon unit tests, 0 failures). The core deletion is correct and complete. Three issues need attention before merge.

### What is correct

1. **`space-task-handlers.ts`** — `isPostApprovalRoutingEnabled` import removed; the `completion_action` intercept (`resumeCompletionActions` branch) fully deleted; the legacy-flag approval branch collapsed to a single `dispatchPostApproval` path. The new `throw` when `spaceRuntimeService` is absent is safe: production always wires the service (`rpc-handlers/index.ts:537`). Tests confirm the throw is unreachable in production.

2. **`space-runtime.ts`** — All six functions confirmed deleted: `resolveCompletionWithActions`, `resumeCompletionActions`, `executeCompletionAction`, `resolveArtifactData`, `buildAwaitingApprovalReason`, `emitTaskAwaitingApproval`, `finalizeResume`. The `awaiting_approval` dedup-key sweep is removed. Both call sites (~line 813 and ~line 1515) now call `dispatchPostApproval` cleanly. The `isPostApprovalRoutingEnabled` import is removed from `space-runtime.ts`.

3. **`space-runtime-service.ts`** — `resumeCompletionActions` public method deleted. `dispatchPostApproval` doc comment updated to remove the flag-gate language.

4. **`space-agent-tools.ts`** — `approve_completion_action` handler and its registration are fully deleted. Grep confirms zero instances of `approve_completion_action` in production source.

5. **`built-in-workflows.ts`** — Zero `completionActions:` entries remain. No references to `MERGE_PR_COMPLETION_ACTION`, `VERIFY_PR_MERGED_COMPLETION_ACTION`, or `PR_MERGE_BASH_SCRIPT`. The `post-approval-merge-template.ts` import is present and correct. The `PR_MERGE_BASH_SCRIPT` in `m94-backfill-workflow-templates.ts` is intentionally frozen in that historical migration.

6. **`post-approval-router.ts`** — `isPostApprovalRoutingEnabled` is retained with updated doc comment (kill-switch only, no fallback path). No production call site reads the flag anymore.

7. **`completion-detector.ts`** — Comment updated from `resolveCompletionWithActions` to `resolveTaskApproval`. No logic changes.

8. **Deleted test files confirmed gone**: `space-runtime-completion-actions.test.ts`, `completion-actions-persistence.test.ts`, `completion-action-executors.test.ts`.

9. **`built-in-workflows.test.ts`** — Adds `completionActions ?? []` equality assertions for Task Dispatcher node. No dangling refs.

### Issues requiring changes

#### Issue 1 — Stale `NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING=0` guards in two test files (misleading)

`space-runtime-completion.test.ts` (lines 226–240) and `space-runtime-tick-loop.test.ts` (lines 178–188) both set `NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING=0` in `beforeEach` with comments stating they are "forcing the kill-switch here so these legacy tests continue to test the legacy path they were written for" and "Deletion of this path happens in PR 4/5". PR 4/5 is this PR. The flag now has zero effect on the runtime (the branch that read it was deleted). The flag guards are harmless but will mislead future readers into thinking there is a legacy fallback.

Both blocks should be removed (the `const SAVED_FLAG`, `beforeEach`, and `afterEach` stanzas). The tests themselves are valid and should be kept.

#### Issue 2 — Misleading kill-switch test comment in `post-approval-routing-integration.test.ts`

Line 498–510 tests `isPostApprovalRoutingEnabled` values. The test comment says "makes the runtime fall back to the legacy completion-actions pipeline (deleted in PR 4/5 — kept alive for this intermediate PR)". This PR IS PR 4/5. The comment implies a live fallback path, which no longer exists. The comment should be updated to reflect the current reality: the function is retained as a diagnostic/kill-switch probe only, with no fallback path.

#### Issue 3 — Dead `CompletionActionExecutedEvent` type and formatter in notification-sink.ts / session-notification-sink.ts

`notification-sink.ts` lines 172–199 still define `CompletionActionExecutedEvent` and include it in the `SpaceNotificationEvent` union (line 295). `session-notification-sink.ts` lines 124–125 still dispatch to `formatCompletionActionExecuted` (lines 405–430). The only emitter of this event was `resumeCompletionActions`, which was deleted in this PR. Nothing in the production source emits `kind: 'completion_action_executed'` anymore.

The PR description says "DB schema and shared types are deliberately untouched — those are deferred to PR 5/5." However these are not DB schema types — they are runtime event types in daemon source. Since the emitter is gone, these are dead code now, not deferred schema. They should be cleaned up in this PR (or, at minimum, a TODO comment should be added noting they are unreachable pending PR 5/5, so they do not confuse the type-checker's exhaustiveness checks).

### Minor stale comment (non-blocking)

`space-agent-tools.ts` line 320-322: the `list_tasks` JSDoc still says tasks are "enriched with `pendingAction` when they are paused at a completion action" — but `enrichTaskWithPendingAction` was deleted in this PR and `list_tasks` no longer calls it. This should be updated.

File: packages/daemon/tests/unit/5-space/runtime/space-runtime-completion.test.ts [hunk 1]
```diff
@@ -207,6 +207,16 @@ class MockTaskAgentManager {
...
+
+	// PR 3/5 introduced post-approval awareness injection via
+	// `injectIntoTaskAgent`. These tests do not assert on delivery; return a
+	// trivial "no session" result so the runtime's best-effort branch is taken.
+	async injectIntoTaskAgent(
+		_taskId: string,
+		_awarenessBody: string
+	): Promise<{ injected: boolean; reason?: string }> {
+		return { injected: false, reason: 'no-session' };
+	}
```
Comment (lsm) [inline packages/daemon/tests/unit/5-space/runtime/space-runtime-completion.test.ts:219]: The stale flag guard block (const SAVED_FLAG at line 233, beforeEach at line 234-236, afterEach at line 237-240) in the parent describe() should be removed. This PR deleted the code branch that read NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING. The flag guard is now a harmless no-op, but the comment 'Deletion of this path happens in PR 4/5' is confusing — this IS PR 4/5. Remove the SAVED_FLAG / beforeEach / afterEach stanza; the tests inside the describe block stay.

File: packages/daemon/tests/unit/5-space/runtime/space-runtime-tick-loop.test.ts [hunk 2]
```diff
@@ -161,6 +161,11 @@ function makeMockTaskAgentManager(
 		rehydrate: overrides.rehydrate ?? (async () => {}),
 		cancelBySessionId: overrides.cancelBySessionId ?? (() => {}),
 		interruptBySessionId: overrides.interruptBySessionId ?? (async () => {}),
+		// PR 3/5 added a post-approval awareness injection via
+		// `injectIntoTaskAgent`. The tick-loop mock is not exercising delivery,
+		// so return a trivial "not injected" result — production treats this as
+		// best-effort anyway.
+		injectIntoTaskAgent: async () => ({ injected: false, reason: 'no-session' }),
```
Comment (lsm) [inline packages/daemon/tests/unit/5-space/runtime/space-runtime-tick-loop.test.ts:168]: Same issue as in space-runtime-completion.test.ts: the NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING=0 beforeEach/afterEach block (lines 178-188 of this file) is now a no-op. PR 4/5 (this PR) deleted the branch in space-runtime.ts that read the flag. The comment says 'PR 4/5 will remove' — that removal has happened. Clean up the SAVED_FLAG / beforeEach / afterEach stanza.

File: packages/daemon/src/lib/space/tools/space-agent-tools.ts [hunk 3]
```diff
@@ -360,10 +358,7 @@ export function createSpaceAgentToolHandlers(config: SpaceAgentToolsConfig) {
 				}));
 				return jsonResult({ success: true, total, tasks: compactTasks });
 			}
-			const enrichedTasks = tasks.map((t) =>
-				enrichTaskWithPendingAction(t, workflowRunRepo, workflowManager)
-			);
-			return jsonResult({ success: true, total, tasks: enrichedTasks });
+			return jsonResult({ success: true, total, tasks });
```
Comment (lsm) [inline packages/daemon/src/lib/space/tools/space-agent-tools.ts:361]: The JSDoc for list_tasks (line 320-322 of this file) still says "enriched with pendingAction when they are paused at a completion action" — but enrichTaskWithPendingAction was deleted in this PR and the non-compact path now returns tasks directly (line 361: return jsonResult({ success: true, total, tasks })). Update or remove the outdated JSDoc sentence.

File: packages/daemon/src/lib/space/runtime/space-runtime.ts [hunk 4]
```diff
@@ -45,23 +43,15 @@ import { Logger } from '../../logger';
...
 	type PostApprovalRouteResult,
 } from './post-approval-router';
 import type { SpaceApprovalSource } from '@neokai/shared';
-import {
-	type CompletionActionExecutionResult,
-	type InstructionActionExecutor,
-	type McpToolExecutor,
-	runMcpCallAction,
-} from './completion-action-executors';
 import {
```
Comment (lsm) [inline packages/daemon/src/lib/space/runtime/space-runtime.ts:55]: The only emitter of completion_action_executed was resumeCompletionActions (deleted in this PR). The CompletionActionExecutedEvent type (notification-sink.ts:172-199), its union membership (notification-sink.ts:295), and the formatCompletionActionExecuted handler (session-notification-sink.ts:124-125, 405-430) are now dead code. The PR description defers "DB schema and shared types" to PR 5/5, but these are runtime event types in daemon source, not DB schema. They should either be deleted here or receive a TODO comment noting the emitter is gone, so exhaustiveness-checking tools do not flag them as live.

**Comment by lsm** [comment]
## PR 4/5 Review — Delete completion-action pipeline + banner consolidation

**Overall:** Solid execution. All required deletions are complete, the single-slot banner renderer with `resolveActiveTaskBanner` is correct and well-tested (exhaustive precedence coverage in `task-banner.test.ts`), `workflow-autonomy.ts` simplification is clean, and CI is fully green. The core work is shippable.

Six issues need fixing before merge — all are quick comment/dead-code cleanups, no logic changes needed.

---

### 🔴 Must fix

**1. Stale "not yet wired" comments in web components**

`packages/web/src/components/space/SpaceTaskPane.tsx` line 37–38:
> `// Not yet written by any runtime path — PR 2 wires it in.`

`packages/web/src/components/space/TaskStatusActions.tsx` lines 12–13 and 42:
> `// No runtime consumer produces this status yet — PR 2 wires it in. ...`
> `// PR 1/5 schema-only: approved has no user-facing enter edge yet.`

These are PR 1/5-vintage comments that are now false — `approved` is actively written by `dispatchPostApproval` and fully wired in the UI as of this PR. A future developer reading these would think the status is still unimplemented. Strip the stale prose, or replace with a concise description of the current state.

---

**2. Dead flag guard in `space-runtime-completion.test.ts` lines 233–239**

```ts
// This suite exercises the legacy `resolveCompletionWithActions` pipeline.
// PR 3/5 flipped the `NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING` flag default
// to ON — which diverts completions through the PostApprovalRouter instead.
// Force the kill-switch here so these legacy tests continue to test the
// legacy path they were written for. Deletion of this path happens in
// PR 4/5; this guard keeps the intermediate state testable.
const SAVED_FLAG = process.env.NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING;
beforeEach(() => { process.env.NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING = '0'; });
afterEach(() => { ... });
```

This IS PR 4/5 — the legacy path is gone. The env-var write is a no-op (no production code reads the flag for routing decisions any more), and the surrounding `describe` comment incorrectly claims this suite tests `resolveCompletionWithActions`. Remove the `SAVED_FLAG` / `beforeEach` / `afterEach` stanza and update the describe-block comment to reflect what the suite actually tests now.

---

**3. Same dead flag guard in `space-runtime-tick-loop.test.ts` lines 178–188**

Same pattern — the `NEOKAI_TASK_AGENT_POST_APPROVAL_ROUTING=0` override and surrounding comment explain an intermediate state that no longer exists. Remove.

---

**4. Stale JSDoc for `list_tasks` in `space-agent-tools.ts` lines 320–322**

```ts
// Tasks in the non-compact output are enriched with `pendingAction` when they are
// paused at a completion action, so consumers can render an approval banner
// without a second fetch.
```

`enrichTaskWithPendingAction` was deleted in this PR — the enrichment no longer happens. Update or remove the stale sentence.

---

### 🟡 Minor (clean up in this PR or add a TODO)

**5. `post-approval-routing-integration.test.ts` line 503**

> `"makes the runtime fall back to the legacy completion-actions pipeline (deleted in PR 4/5 — kept alive for this intermediate PR)"`

The "kept alive for this intermediate PR" clause is now wrong (this IS PR 4/5 and the pipeline is gone). The test itself is still valid (it checks `isPostApprovalRoutingEnabled` parse behaviour), but the comment needs updating.

---

**6. Dead `CompletionActionExecutedEvent` in daemon runtime**

`packages/daemon/src/lib/space/runtime/notification-sink.ts` — `CompletionActionExecutedEvent` interface + union membership at line 295. `packages/daemon/src/lib/space/runtime/session-notification-sink.ts` — `case 'completion_action_executed'` handler and `formatCompletionActionExecuted` function. The only emitter (`resumeCompletionActions`) was deleted in this PR. These are daemon-local runtime event types (not the shared `CompletionAction` types deferred to PR 5), so they're in scope for deletion here. Either delete them now or add an explicit `// TODO(PR5): remove CompletionActionExecutedEvent — emitter deleted in PR4` comment so they don't linger indefinitely.

---

### ✅ Verified correct

- All seven target functions in `space-runtime.ts` deleted; both call sites now call `dispatchPostApproval` directly.
- `approve_completion_action` handler and registration cleanly removed from `space-agent-tools.ts`.
- `completion_action` RPC intercept and legacy flag-branch fully removed from `space-task-handlers.ts`. Error throw on absent `spaceRuntimeService` is safe (always wired in production).
- All `completionActions: [...]` workflow entries deleted; migration test fixtures untouched (correct — historical).
- `isWorkflowAutoClosingAtLevel` replacement is correct: `level >= (wf.completionAutonomyLevel ?? 5)`. All callers updated.
- `resolveActiveTaskBanner` precedence is exact (`blocked > post_approval_blocked > task_completion_pending > gate_pending > null`). Tests cover all combos including whitespace trim and legacy `completion_action` passthrough.
- All three banner components compose `InlineStatusBanner`; no inline content exceeds one line.
- `post-approval-banner-minimal.e2e.ts` and `post-approval-no-route.e2e.ts` are fully implemented (not stubs). Skipped E2E stubs contain detailed rationale and step outlines.
- `bun run check` clean; all required CI checks green.

File: packages/daemon/tests/unit/5-space/runtime/space-runtime-completion.test.ts (hunk 1 above)
Comment (lsm) [reply packages/daemon/tests/unit/5-space/runtime/space-runtime-completion.test.ts:219]: Fixed in 51e5ee0ba — removed the SAVED_FLAG / beforeEach / afterEach stanza and rewrote the describe-block comment to describe what the suite tests now (completion detection + transitions + post-approval router dispatch on `approved`).

File: packages/daemon/tests/unit/5-space/runtime/space-runtime-tick-loop.test.ts (hunk 2 above)
Comment (lsm) [reply packages/daemon/tests/unit/5-space/runtime/space-runtime-tick-loop.test.ts:168]: Fixed in 51e5ee0ba — removed the SAVED_FLAG / beforeEach / afterEach block and updated the describe-block comment to reflect that `dispatchPostApproval` is the only routing path now.

File: packages/daemon/src/lib/space/tools/space-agent-tools.ts (hunk 3 above)
Comment (lsm) [reply packages/daemon/src/lib/space/tools/space-agent-tools.ts:361]: Fixed in 51e5ee0ba — replaced the stale 'enriched with pendingAction' sentence with a description of the `compact` projection (id, title, status, priority, createdAt) and the default full-row return.

File: packages/daemon/src/lib/space/runtime/space-runtime.ts (hunk 4 above)
Comment (lsm) [reply packages/daemon/src/lib/space/runtime/space-runtime.ts:55]: Fixed in 51e5ee0ba — deleted `CompletionActionExecutedEvent` from notification-sink.ts (interface + union membership), the `case 'completion_action_executed'` and `formatCompletionActionExecuted` from session-notification-sink.ts, and the corresponding `describe('formatEventMessage — completion_action_executed')` block from session-notification-sink.test.ts. Also updated the lingering `completion_action audit trail` reference in the WorkflowRunReopenedEvent JSDoc to talk about post-approval audit artifacts instead.

**Comment by lsm** [comment]
All six review items addressed in 51e5ee0ba:

**Items 1–4 + 6** (inline-commented threads): replied on each. Summary:

- **SpaceTaskPane.tsx** — replaced the stale 'PR 2 wires it in' comment with a one-liner describing what `approved` actually means now ("post-approval staging status\: tasks land here when an agent calls `approve_task`, then PostApprovalRouter dispatches the follow-up").
- **TaskStatusActions.tsx** — both stale comments rewritten. The `approved` row in `VALID_TASK_TRANSITIONS` now documents the conservative escape-hatch transition set in current terms; the `approved->X` label block now explains that these labels render only for manual transitions (router failure / admin intervention).
- **space-runtime-completion.test.ts + space-runtime-tick-loop.test.ts** — `SAVED_FLAG` / `beforeEach` / `afterEach` env-guards deleted; describe-block comments rewritten to describe what each suite tests today.
- **space-agent-tools.ts list_tasks JSDoc** — replaced the stale `pendingAction` enrichment sentence with a description of the `compact` projection.
- **post-approval-routing-integration.test.ts line 503** — comment now states the parsing contract (the flag's only remaining consumer is SpaceRuntime feature-gating; legacy completion-actions pipeline is gone) instead of "kept alive for this intermediate PR".

**Item 6** (CompletionActionExecutedEvent dead code): deleted in this PR rather than deferred. Removed the interface + union membership in notification-sink.ts, the dispatch case + formatter in session-notification-sink.ts, and the corresponding test block in session-notification-sink.test.ts. Also updated the `completion_action audit trail` reference in the WorkflowRunReopenedEvent JSDoc.

Local tests: `bun run check` clean; daemon 5-space unit tests 2441/2441; web vitest 7231/7231; targeted post-approval suites 59/59.

CI is running on the new commit.

**Comment by lsm** [comment]
All requested changes addressed cleanly in `51e5ee0ba`. Verified:

- **SpaceTaskPane.tsx** — stale "Not yet written by any runtime path" comment replaced with accurate one-liner describing `approved` → `dispatchPostApproval` → PostApprovalRouter.
- **TaskStatusActions.tsx** — both stale PR 1/5 comments rewritten; conservative escape-hatch rationale preserved and current.
- **space-runtime-completion.test.ts** — dead flag guard deleted; describe-block comment now correctly describes what the suite tests (completion detection + PostApprovalRouter dispatch).
- **space-runtime-tick-loop.test.ts** — same flag guard gone.
- **space-agent-tools.ts list_tasks JSDoc** — stale `pendingAction` enrichment sentence removed; replaced with accurate compact-projection note.
- **post-approval-routing-integration.test.ts** — "kept alive for this intermediate PR" language removed; comment now describes the parsing contract.
- **CompletionActionExecutedEvent** — fully deleted (interface + union in `notification-sink.ts`; case + formatter in `session-notification-sink.ts`; corresponding test block removed). Zero residual references in daemon src.

CI: 38/38 required checks pass on latest commit. mergeStateStatus: CLEAN. ✅
